magenta-lib/src/main/scala/magenta/deployment_type/S3.scala

```diff
@@ -106,18 +107,18 @@ object S3 extends DeploymentType with S3AclParams {
         assert(data.isDefined, s"Cannot find resource value for ${bucketResource(pkg)} (${pkg.apps.head} in ${parameters.stage.name})")
         data.get.value
       }
+      val prefix:String = S3UploadV2.prefixGenerator(
+        stack = if (prefixStack(pkg)) Some(stack) else None,
+        stage = if (prefixStage(pkg)) Some(parameters.stage) else None,
+        packageName = if (prefixPackage(pkg)) Some(pkg.name) else None
+      )
```

Comment:
It should be identical. The problem is that it also needs the parameters from the class, so it's quite difficult to dry out without passing in seven parameters. I could extract it into a param trait.